cedar-policy-core/src/ast/request.rs

```diff
@@ -309,28 +316,28 @@ impl Context {
             .from_json_file(json)
     }
 
-    /// Iterate over the (key, value) pairs in the `Context`; or return `None`
-    /// if the `Context` is purely unknown
+    /// Get an iterator over the (key, value) pairs in the `Context`, cloning
+    /// only if necessary.
     //
-    // PANIC SAFETY: This is safe due to the invariant on `self.context`, `self.context` must always be a record
-    pub fn iter<'s>(&'s self) -> Option<Box<dyn Iterator<Item = (&SmolStr, PartialValue)> + 's>> {
+    // PANIC SAFETY: This is safe due to the invariant (ContextRecord) on `self.context`
+    pub fn values(self) -> Box<dyn Iterator<Item = (SmolStr, PartialValue)>> {
```

Comment:
Should we call this `.into_iter()` instead?  Maybe even have it be an implementation of the `IntoIterator` trait?

Reply:
I'll try this out tomorrow

Reply:
Latest version adds an implementation of `IntoIterator`. I don't think we can provide an `.iter()` function that doesn't perform cloning atm. I think the best solution would be to address #541 by storing the context as a hash map instead of a record expression.

Reply:
Nicely done :)

Want to add a comment to #541 with your suggestion for implementation?